src/mgr/ActivePyModules.cc

```diff
@@ -1516,16 +1516,41 @@ void ActivePyModules::register_client(std::string_view name, std::string addrs)
   entity_addrvec_t addrv;
   addrv.parse(addrs.data());
 
-  dout(7) << "registering msgr client handle " << addrv << dendl;
-  py_module_registry.register_client(name, std::move(addrv));
+  dout(7) << "registering msgr client handle [" << name << "] " << addrv << dendl;
+  epoch_t e;
+  cluster_state.with_mgrmap([&e](const MgrMap& m) {
+    e = m.get_epoch();
+  });
```

Comment:
```suggestion
if (cluster_state.is_client_in_mgrmap(name, addrv, &e)) {
  return;
}
```

We can avoid inserting a duplicate. Additionally, if we deduplicate addresses in the future in `PyModuleRegistry`, it avoids the situation:
- get current mgrmap
- add duplicate address, dropped by pymoduleregistry
- wait for next mgrmap but no new addresses are sent in the beacon so new mgrmap is forthcoming...
- stuck waiting for a really long time...

Reply:
Nice! I'll make this change.